src/main/sensors/battery.c

```diff
@@ -543,3 +543,16 @@ int32_t getMAhDrawn(void)
 {
     return currentMeter.mAhDrawn;
 }
+
+float getWhDrawn(void)
+{
+    static int32_t mAhDrawnPrev = 0;
+    static float wattHoursDrawn = 0.0; 
+    const int32_t mAhDrawnCurrent = getMAhDrawn();
+
+    wattHoursDrawn += voltageMeter.displayFiltered * (mAhDrawnCurrent - mAhDrawnPrev) / 100000.0f;
+
+    mAhDrawnPrev = mAhDrawnCurrent;
+
+    return wattHoursDrawn;
+}
```

Comment:
This is a getter function. It must only return the calculated value, but not calculate the value itself. The calculation needs to be somewhere in a loop where it gets updated automatically. Then you can call `getWhDrawn()` whenever you need and how often you want to get the temporary result for Wh used

`getWhDrawn()` needs to look more like this:

```c
float getWhDrawn(void)
{
    return wattHoursDrawn;
}
```